ui/vendor/_directional.scss

```diff
@@ -0,0 +1,63 @@
+// directional-scss | Author: Tyson Matanich (http://matanich.com), 2013 | License: MIT
+$dir: ltr !default;
+
+// Default $dir if not valid
+@if $dir != ltr and $dir != rtl {
+    $dir: ltr;
+}
+
+@function if-ltr($if, $else: null) {
+  @if $dir != rtl {
+    @return $if;
+  }
+  @else {
+    @return $else;
+  }
+}
+
+@function if-rtl($if, $else: null) {
+  @return if-ltr($else, $if);
+}
+
+$left: if-ltr(left, right);
+$right: if-ltr(right, left);
```

Comment:
A few more challenges to the approach of keeping `left` as `left` in all scenario's. Take the form input where someone adds `[dir=rtl]` but keeps the `slds-input-has-icon--left`:

```
<div class="slds-form-element">
  <label class="slds-form-element__label" for="text-input-01">Input Label</label>
  <div class="slds-form-element__control slds-input-has-icon slds-input-has-icon--left">
    <svg class="slds-input__icon slds-icon-text-default" aria-hidden="true">
      <use xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#search"></use>
    </svg>
    <input type="text" id="text-input-01" class="slds-input" placeholder="Placeholder Text" />
  </div>
</div>
```

The icon will stay on the left, but by default the text will be sent to the right:

![image](https://cloud.githubusercontent.com/assets/415690/24859503/b0b8e47e-1da5-11e7-9971-32b136da9b9b.png)

Unless you then add `direction: ltr` explicitly to it- which is of course an option. But I think the biggest problem comes from having to reset padding's, etc when using `start`, `end`. You end up with rules like this:

```
&--left,
  &--start {
    .slds-input__icon {
      left: ($spacing-medium + $spacing-xx-small);
    }

    .slds-input,
    .slds-input--bare {
      padding-left: $spacing-xx-large;
    }
  }

[dir=rtl] {
    &--start {
      .slds-input__icon {
        right: ($spacing-medium + $spacing-xx-small);
        left: 0;
      }

      .slds-input,
      .slds-input--bare {
        padding-right: $spacing-xx-large;
        padding-left: 0;
      }
    }
}
```
Aside from it being ugly, it becomes a problem when someone has added `slds-p-left--large` as well - the new rule would take precedence due to it's specificity being higher.